ichnaea/log.py

```diff
@@ -179,54 +179,78 @@ def configure_stats():
 
 
 def log_tween_factory(handler, registry):
-    """A logging tween, doing automatic statsd and raven collection."""
+    """A logging tween, handling collection of stats, exceptions, and a request log."""
 
-    def log_tween(request):
-        if request.path in registry.skip_logging or request.path.startswith("/static"):
-            # shortcut handling for static assets
-            try:
-                return handler(request)
-            except HTTPException:
-                # don't capture exceptions for normal responses
-                raise
-            except Exception:
-                registry.raven_client.captureException()
-                raise
+    local_dev_env = settings("local_dev_env")
 
+    def log_tween(request):
+        """Time a request, emit metrics and log results, with exception handling."""
         start = time.time()
-        statsd_tags = [
-            # Convert a URI to a statsd acceptable metric name
-            "path:%s" % request.path.replace("/", ".").lstrip(".").replace("@", "-"),
-            "method:%s" % request.method.lower(),
-        ]
-
-        def timer_send():
-            duration = int(round((time.time() - start) * 1000))
-            METRICS.timing("request.timing", duration, tags=statsd_tags)
+        structlog.threadlocal.clear_threadlocal()
+        structlog.threadlocal.bind_threadlocal(
+            http_method=request.method, http_path=request.path
+        )
+        # Skip detailed logging and capturing for static assets, either in
+        # /static or paths like /robots.txt
+        full_logs = not (
+            request.path in registry.skip_logging or request.path.startswith("/static")
+        )
 
-        def counter_send(status_code):
-            METRICS.incr("request", tags=statsd_tags + ["status:%s" % status_code])
+        def record_response(status_code):
+            """Time request, (maybe) emit metrics, and (maybe) log this request.
+
+            For static assets, metrics are skipped, and logs are skipped unless
+            we're in the development environment.
+            """
+            duration = time.time() - start
+
+            if full_logs:
+                # Emit a request.timing and a request metric
+                duration_ms = int(round(duration * 1000))
+                statsd_tags = [
+                    # Convert a URI to a statsd acceptable metric name
+                    "path:%s"
+                    % request.path.replace("/", ".").lstrip(".").replace("@", "-"),
+                    "method:%s" % request.method.lower(),
```

Comment:
Markus has a utility function for fixing tag values:

https://markus.readthedocs.io/en/latest/usage.html#markus.utils.generate_tag

Does that help here?